core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -15,7 +15,16 @@
 from settings import Settings
 
 
-class ArduPilotManager:
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
```

Comment:
I would recommend to move the singleton class to a new class to make it more flexible and useful, it also makes sense since you are using a dict of instances and not a single instance.

Reply:
Makes sense.